### PR TITLE
Inbox ID updates

### PR DESCRIPTION
**Pull request by neekolas**
## tl;dr

- Adds new `UpdateGroupMembershipData` intent, which will be used for adding/removing members
- Adds `creator_inbox_id` to `GroupMetadataV1` (but doesn't yet remove `creator_account_address`)

### CODE REVIEW

File: proto/mls/message_contents/group_metadata.proto [hunk 1]
```diff
@@ -9,8 +9,10 @@ option java_package = "org.xmtp.proto.mls.message.contents";
 // Parent message for group metadata
 message GroupMetadataV1 {
   ConversationType conversation_type = 1;
+  // This will be removed soon
```
Comment (tuddman) [inline proto/mls/message_contents/group_metadata.proto:12]: Absolute confidence it will be 😛

**Review by tuddman — APPROVED** [review]

File: proto/mls/database/intents.proto [hunk 2]
```diff
@@ -61,6 +61,22 @@ message RemoveMembersData {
   }
 }
 
+// The data required to make a commit that updates group membership
+// Handles both Add and Remove actions
+message UpdateGroupMembershipData {
+  // V1 of UpdateGroupMembershipPublishData
+  message V1 {
+    // Contains delta of membership updates that need to be applied
+    map<string, uint64> membership_updates = 1;
```
Comment (37ng) [inline proto/mls/database/intents.proto:70]: What are the key, value of this map? (Might be good to document it)

File: proto/mls/database/intents.proto [hunk 3]
```diff
@@ -61,6 +61,22 @@ message RemoveMembersData {
...
+    // Contains the list of members that will be removed
+    repeated string removed_members = 2;
```
Comment (37ng) [inline proto/mls/database/intents.proto:72]: If the `membership_updates` contains delta, what are `removed_member` for?

Comment (neekolas) [reply proto/mls/database/intents.proto:72]: I thought about having an update that has a `sequence_id` of 0 be treated as a remove, but it felt a little hacky.

**Pull request merged**